viewer/packages/360-images/src/entity/Image360Entity.ts

```diff
@@ -37,111 +24,89 @@ export class Image360Entity implements Image360 {
-      }
-    );
+  public setOpacity(alpha: number): void {
+    this._revisions.forEach(revision => (revision.image360Visualization.opacity = alpha));
+  }
 
-    const fullResolutionCompleted = fullResolutionFaces
-      .catch(e => {
-        return Promise.reject(e);
-      })
-      .then(
-        () => {
-          return Promise.resolve();
-        },
-        reason => {
-          return Promise.reject(reason);
-        }
-      );
+  /**
+   * List all available revisions.
+   * */
+  public list360ImageRevisions(): Image360RevisionEntity[] {
+    return this._revisions;
+  }
 
-    return { firstCompleted, fullResolutionCompleted };
+  /**
+   * Will reload the entity with images from the new revision.
+   * Resolves once loading is complete. Rejects if revision could not be changed.
+   * If the entity is not entered/visible the promise will be resolved instantly.
+   *
+   * @param revision The revision to load
+   * @returns Promise for when revision has either been updated or it failed to change.
+   */
+  public changeRevision(revision: Image360RevisionEntity): Promise<void> {
+    return this._reloadImage(this, revision);
   }
 
   /**
-   * Apply full resolution textures to the image360VisualzationBox. This has no effect if full resolution has already been applied.
+   * Get the revision that is currently loaded for this entry.
+   * @returns Returns the active revision.
    */
-  public async applyFullResolution(): Promise<void> {
-    if (this._getFullResolutionTextures) {
-      const result = await this._getFullResolutionTextures.catch(() => {});
-      if (result) {
-        this._image360VisualzationBox.loadImages(await result.textures);
-        this._getFullResolutionTextures = undefined;
-      }
-    }
+  public getActiveRevision(): Image360RevisionEntity {
+    return this._activeRevision;
+  }
+
+  public setActiveRevision(revision: Image360RevisionEntity): void {
+    this._activeRevision = revision;
+  }
+
+  public getMostRecentRevision(): Image360RevisionEntity {
+    return this._revisions[0];
   }
 
   /**
-   * Drops the GPU resources for the 360 image
-   * the icon will be preserved.
+   * Get the revision closest to the provided date.
+   * If all revisions are undated the first available revison is returned.
    */
-  public unload360Image(): void {
-    this._getFullResolutionTextures = undefined;
-    this._image360VisualzationBox.unloadImages();
+  public getRevisionClosestToDate(date: Date): Image360RevisionEntity {
+    const dateAsNumber = date.getTime();
+    const closest = this._revisions.reduce(
+      (closest, revision) => {
+        if (revision.date) {
+          const difference = Math.abs(revision.date.getTime() - dateAsNumber);
+          if (difference < closest.difference) return { revision: revision, difference };
+        }
+        return closest;
+      },
+      { revision: this.getMostRecentRevision(), difference: Number.POSITIVE_INFINITY }
+    );
+    return closest.revision;
```

Comment:
Think this can be made cleaner with something like this (note that I havn't tested this):
```ts
  public getRevisionClosestToDate(date: Date): Image360RevisionEntity {
    const dateAsNumber = date.getTime();
    const datedRevisions = this.revisions.filter(revision => revision.date !== undefined);
    const closestDatedRevision = minBy(datedRevisions, revision => Math.abs(revision.date.getTime() - dateAsNumber));
    return closestDatedRevision ?? this.getMostRecentRevision();
```